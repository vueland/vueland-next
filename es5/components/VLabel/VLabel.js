"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VLabel = void 0;

require("../../../src/components/VLabel/VLabel.scss");

var _vue = require("vue");

var _helpers = require("../../helpers");

var _useColors2 = require("../../effects/use-colors");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var VLabel = (0, _vue.defineComponent)({
  name: 'v-label',
  props: _objectSpread({
    absolute: Boolean,
    disabled: Boolean,
    focused: Boolean,
    onField: Boolean,
    left: {
      type: [Number, String],
      "default": 0
    },
    right: {
      type: [Number, String],
      "default": 'auto'
    },
    hasState: Boolean
  }, (0, _useColors2.colorProps)()),
  setup: function setup(props, _ref) {
    var slots = _ref.slots;

    var _useColors = (0, _useColors2.useColors)(),
        setTextColor = _useColors.setTextColor;

    var isActive = (0, _vue.computed)(function () {
      return !!props.hasState || !!props.focused;
    });
    var classes = (0, _vue.computed)(function () {
      return {
        'v-label': true,
        'v-label--active': isActive.value,
        'v-label--on-field': props.onField,
        'v-label--has-state': props.hasState,
        'v-label--is-disabled': !!props.disabled
      };
    });

    function genPropsData() {
      return {
        "class": classes.value,
        style: {
          left: (0, _helpers.convertToUnit)(props.left),
          right: (0, _helpers.convertToUnit)(props.right),
          position: props.absolute ? 'absolute' : 'relative'
        }
      };
    }

    return function () {
      var propsData = genPropsData();
      return (0, _vue.h)('label', props.color ? setTextColor(props.color, propsData) : propsData, slots["default"] && slots["default"]());
    };
  }
});
exports.VLabel = VLabel;
//# sourceMappingURL=VLabel.js.map