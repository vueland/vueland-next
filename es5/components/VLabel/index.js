"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _VLabel = require("./VLabel");

Object.keys(_VLabel).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _VLabel[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _VLabel[key];
    }
  });
});
//# sourceMappingURL=index.js.map