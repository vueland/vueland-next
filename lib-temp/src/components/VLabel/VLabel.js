import { defineComponent, h, computed } from 'vue';
import { useColors, colorProps } from '../../composable/use-colors';
export const VLabel = defineComponent({
    name: 'v-label',
    props: {
        disabled: Boolean,
        focused: Boolean,
        ...colorProps(),
    },
    setup(props, { slots }) {
        const { setTextClassNameColor, setTextCssColor } = useColors();
        const classes = computed(() => ({
            'v-label': true,
            'v-label--disabled': props.disabled,
            'v-label--focused': props.focused,
            ...(!props.disabled ? setTextClassNameColor(props.color) : {}),
        }));
        const styles = computed(() => ({
            ...(!props.disabled ? setTextCssColor(props.color) : {}),
        }));
        return () => h('label', {
            class: classes.value,
            style: styles.value,
        }, {
            default: () => slots.default && slots.default(),
        });
    },
});
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVkxhYmVsLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiLi4vLi4vLi4vLi4vc3JjL2NvbXBvbmVudHMvVkxhYmVsL1ZMYWJlbC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUUsZUFBZSxFQUFFLENBQUMsRUFBRSxRQUFRLEVBQUUsTUFBTSxLQUFLLENBQUE7QUFDbEQsT0FBTyxFQUFFLFNBQVMsRUFBRSxVQUFVLEVBQUUsTUFBTSw2QkFBNkIsQ0FBQTtBQUVuRSxNQUFNLENBQUMsTUFBTSxNQUFNLEdBQUcsZUFBZSxDQUFDO0lBQ3BDLElBQUksRUFBRSxTQUFTO0lBQ2YsS0FBSyxFQUFFO1FBQ0wsUUFBUSxFQUFFLE9BQU87UUFDakIsT0FBTyxFQUFFLE9BQU87UUFDaEIsR0FBRyxVQUFVLEVBQUU7S0FDaEI7SUFDRCxLQUFLLENBQUMsS0FBSyxFQUFFLEVBQUUsS0FBSyxFQUFFO1FBQ3BCLE1BQU0sRUFBRSxxQkFBcUIsRUFBRSxlQUFlLEVBQUUsR0FBRyxTQUFTLEVBQUUsQ0FBQTtRQUU5RCxNQUFNLE9BQU8sR0FBRyxRQUFRLENBQTBCLEdBQUcsRUFBRSxDQUFDLENBQUM7WUFDdkQsU0FBUyxFQUFFLElBQUk7WUFDZixtQkFBbUIsRUFBRSxLQUFLLENBQUMsUUFBUTtZQUNuQyxrQkFBa0IsRUFBRSxLQUFLLENBQUMsT0FBTztZQUNqQyxHQUFHLENBQUMsQ0FBQyxLQUFLLENBQUMsUUFBUSxDQUFDLENBQUMsQ0FBQyxxQkFBcUIsQ0FBQyxLQUFLLENBQUMsS0FBSyxDQUFDLENBQUMsQ0FBQyxDQUFDLEVBQUUsQ0FBQztTQUMvRCxDQUFDLENBQUMsQ0FBQTtRQUVILE1BQU0sTUFBTSxHQUFHLFFBQVEsQ0FBeUIsR0FBRyxFQUFFLENBQUMsQ0FBQztZQUNyRCxHQUFHLENBQUMsQ0FBQyxLQUFLLENBQUMsUUFBUSxDQUFDLENBQUMsQ0FBQyxlQUFlLENBQUMsS0FBSyxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUMsQ0FBQyxFQUFFLENBQUM7U0FDekQsQ0FBQyxDQUFDLENBQUE7UUFFSCxPQUFPLEdBQUcsRUFBRSxDQUNWLENBQUMsQ0FDQyxPQUFPLEVBQ1A7WUFDRSxLQUFLLEVBQUUsT0FBTyxDQUFDLEtBQUs7WUFDcEIsS0FBSyxFQUFFLE1BQU0sQ0FBQyxLQUFLO1NBQ3BCLEVBQ0Q7WUFDRSxPQUFPLEVBQUUsR0FBRyxFQUFFLENBQUMsS0FBSyxDQUFDLE9BQU8sSUFBSSxLQUFLLENBQUMsT0FBTyxFQUFFO1NBQ2hELENBQ0YsQ0FBQTtJQUNMLENBQUM7Q0FDRixDQUFDLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBkZWZpbmVDb21wb25lbnQsIGgsIGNvbXB1dGVkIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdXNlQ29sb3JzLCBjb2xvclByb3BzIH0gZnJvbSAnLi4vLi4vY29tcG9zYWJsZS91c2UtY29sb3JzJ1xuXG5leHBvcnQgY29uc3QgVkxhYmVsID0gZGVmaW5lQ29tcG9uZW50KHtcbiAgbmFtZTogJ3YtbGFiZWwnLFxuICBwcm9wczoge1xuICAgIGRpc2FibGVkOiBCb29sZWFuLFxuICAgIGZvY3VzZWQ6IEJvb2xlYW4sXG4gICAgLi4uY29sb3JQcm9wcygpLFxuICB9LFxuICBzZXR1cChwcm9wcywgeyBzbG90cyB9KSB7XG4gICAgY29uc3QgeyBzZXRUZXh0Q2xhc3NOYW1lQ29sb3IsIHNldFRleHRDc3NDb2xvciB9ID0gdXNlQ29sb3JzKClcblxuICAgIGNvbnN0IGNsYXNzZXMgPSBjb21wdXRlZDxSZWNvcmQ8c3RyaW5nLCBib29sZWFuPj4oKCkgPT4gKHtcbiAgICAgICd2LWxhYmVsJzogdHJ1ZSxcbiAgICAgICd2LWxhYmVsLS1kaXNhYmxlZCc6IHByb3BzLmRpc2FibGVkLFxuICAgICAgJ3YtbGFiZWwtLWZvY3VzZWQnOiBwcm9wcy5mb2N1c2VkLFxuICAgICAgLi4uKCFwcm9wcy5kaXNhYmxlZCA/IHNldFRleHRDbGFzc05hbWVDb2xvcihwcm9wcy5jb2xvcikgOiB7fSksXG4gICAgfSkpXG5cbiAgICBjb25zdCBzdHlsZXMgPSBjb21wdXRlZDxSZWNvcmQ8c3RyaW5nLCBzdHJpbmc+PigoKSA9PiAoe1xuICAgICAgLi4uKCFwcm9wcy5kaXNhYmxlZCA/IHNldFRleHRDc3NDb2xvcihwcm9wcy5jb2xvcikgOiB7fSksXG4gICAgfSkpXG5cbiAgICByZXR1cm4gKCkgPT5cbiAgICAgIGgoXG4gICAgICAgICdsYWJlbCcsXG4gICAgICAgIHtcbiAgICAgICAgICBjbGFzczogY2xhc3Nlcy52YWx1ZSxcbiAgICAgICAgICBzdHlsZTogc3R5bGVzLnZhbHVlLFxuICAgICAgICB9LFxuICAgICAgICB7XG4gICAgICAgICAgZGVmYXVsdDogKCkgPT4gc2xvdHMuZGVmYXVsdCAmJiBzbG90cy5kZWZhdWx0KCksXG4gICAgICAgIH1cbiAgICAgIClcbiAgfSxcbn0pXG4iXX0=