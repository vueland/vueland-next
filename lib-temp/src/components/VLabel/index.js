export * from "./VLabel";
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WTGFiZWwvaW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsY0FBYyxVQUFVLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgKiBmcm9tIFwiLi9WTGFiZWxcIlxuIl19