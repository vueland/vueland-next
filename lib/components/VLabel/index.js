export * from "./VLabel";
//# sourceMappingURL=index.js.map