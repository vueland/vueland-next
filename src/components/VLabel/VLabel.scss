@import "../../styles/scss/modules/global";

.v-label {
  color: currentColor;
  transition: .4s left $primary-transition;

  &--on-input {
    position: absolute;
    left: 0;
  }

  &--focused {
    left: 10px;
  }

  &--disabled {
    color: map-get($disabled, 'base');
  }
}
