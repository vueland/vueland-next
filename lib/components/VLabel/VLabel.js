import "../../../src/components/VLabel/VLabel.scss";
import { defineComponent, h, computed } from 'vue';
import { convertToUnit } from '../../helpers';
import { colorProps, useColors } from '../../effects/use-colors';
const labelProps = {
  absolute: Boolean,
  disabled: Boolean,
  focused: Boolean,
  for: String,
  left: {
    type: [Number, String],
    default: 0
  },
  right: {
    type: [Number, String],
    default: 'auto'
  },
  hasState: Boolean,
  ...colorProps()
};
export const VLabel = defineComponent({
  name: 'v-label',
  props: labelProps,

  setup(props, {
    slots
  }) {
    const {
      setTextColor
    } = useColors();
    const classes = computed(() => {
      return {
        'v-label': true,
        'v-label--active': !!props.hasState || !!props.focused,
        'v-label--is-disabled': !!props.disabled
      };
    });

    const genDataProps = () => {
      return {
        class: { ...classes.value
        },
        style: {
          left: convertToUnit(props.left),
          right: convertToUnit(props.right),
          position: props.absolute ? 'absolute' : 'relative'
        }
      };
    };

    return h('label', setTextColor(props.color, genDataProps()), slots.default && slots.default());
  }

});
//# sourceMappingURL=VLabel.js.map