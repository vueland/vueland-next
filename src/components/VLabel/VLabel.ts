// Styles
import './VLabel.scss'

// Vue API
import {
  h,
  computed,
  defineComponent
} from 'vue'

// Helpers
import { convertToUnit } from '../../helpers'

// Effects
import { colorProps, useColors } from '../../effects/use-colors'

const labelProps = {
  absolute: Boolean,
  disabled: Boolean,
  focused: Boolean,
  for: String,
  left: {
    type: [Number, String],
    default: 0,
  },
  right: {
    type: [Number, String],
    default: 'auto',
  },
  hasState: Boolean,
  ...colorProps(),
}

export const VLabel = defineComponent({
  name: 'v-label',
  props: labelProps,

  setup(props, { slots }) {
    const { setTextColor } = useColors()

    const isActive = computed(() => {
      return !!props.hasState || !!props.focused
    })

    const classes = computed<Record<string, boolean>>(
      () => {
        return {
          'v-label': true,
          'v-label--active': isActive.value,
          'v-label--is-disabled': !!props.disabled,
        }
      },
    )

    const genDataProps = () => {
      return {
        class: {
          ...classes.value,
        },
        style: {
          left: convertToUnit(props.left),
          right: convertToUnit(props.right),
          position: props.absolute ? 'absolute' : 'relative',
        },
      }
    }

    return h(
      'label',
      setTextColor(props.color!, genDataProps()),
      slots.default && slots.default(),
    )
  },
})
