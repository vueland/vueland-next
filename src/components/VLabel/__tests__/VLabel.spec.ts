import { mount, VueWrapper } from "@vue/test-utils";
import { VLabel } from "../VLabel";

describe("VLabel", () => {
  let mountFunction: (options?: any) => VueWrapper<any>;

  beforeEach(() => {
    mountFunction = (options = {}) => mount(VLabel, { ...options });
  });

  it("should mount component and match snapshot", () => {
    const cmp = mountFunction();

    expect(cmp.html()).toMatchSnapshot();
  });
});
