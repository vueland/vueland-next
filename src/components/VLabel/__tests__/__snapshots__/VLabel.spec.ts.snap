// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VLabel should mount component and match snapshot 1`] = `
<label class="v-label">
</label>
`;
